core/src/main/java/gyro/core/resource/DiffableType.java

```diff
@@ -156,7 +157,12 @@ public D newExternal(RootScope root, Object id) {
         diffable.external = true;
         diffable.scope = new DiffableScope(root, null);
 
+        if (idField == null) {
+            throw new Bug(String.format("%s is missing having @Id annotation on its reference field.", diffableClass.getName()));
```

Comment:
Made a slight tweak to the error message.

```suggestion
            throw new Bug(String.format("%s is missing @Id annotation.", diffableClass.getName()));
```